### PR TITLE
Remove deprecated `useDisMax` from MultiMatchQuery

### PR DESCRIPTION
**Pull request by cbuescher**
The getters and setters for useDisMax() have been deprecated since at least 6.0,
also there hasn't been any reference to the query parameter in the
documentation. Removing it from the builder and tests and replacing it with
`tieBreaker(1.0f)` where necessary.

### CODE REVIEW

**Review by jtibshirani — APPROVED** [review]
Looks good to me. To be really safe we could deprecate the `use_dis_max` REST parameter in 6.6? The parameter may still pop up in some places -- for example, it looks like it's being used in `JdbcCsvSpecIT`.

File: server/src/main/java/org/elasticsearch/index/query/MultiMatchQueryBuilder.java [hunk 1]
```diff
@@ -256,7 +256,9 @@ protected void doWriteTo(StreamOutput out) throws IOException {
         out.writeVInt(maxExpansions);
         out.writeOptionalString(minimumShouldMatch);
         out.writeOptionalString(fuzzyRewrite);
-        out.writeOptionalBoolean(useDisMax);
+        if (out.getVersion().before(Version.V_7_0_0)) {
+            out.writeOptionalBoolean(null); //
```
[the remark this answers is not in the document]
Comment (cbuescher) [reply server/src/main/java/org/elasticsearch/index/query/MultiMatchQueryBuilder.java]: Sorry, leftover. Will remove.

**Comment by cbuescher** [comment]
> To be really safe we could deprecate the `use_dis_max` REST parameter in 6.6? The parameter may still pop up in some places -- for example, it looks like it's being used in `JdbcCsvSpecIT`.

I misses this usage in some sql qa tests. Adding deprecation warnings for the rest usages in 6.x are a good idea but I'd say we can still remove it in 7.0 since at least for MultiMatch this parameter hasn't been documented at last since 5.6 (there was some docs for use in Query string query, but there also has been a note in `docs/reference/migration/migrate_6_0/search.asciidoc` that its use in `query_string` has been removed).

**Comment by cbuescher** [comment]
@elasticmachien test this please

**Pull request merged**